e2e/rook-test-framework/rook-infra-manager/rook_test_infra_manager.go

```diff
@@ -0,0 +1,323 @@
+			k8sVersion:      version,
+		}
+	})
+
+	return nil, r
+}
+
+func getCurrentDirectory() string {
+	_, filename, _, ok := runtime.Caller(0)
+	if !ok {
+		panic("No caller information")
+	}
+
+	return path.Dir(filename)
+}
+
+//Gets the platform the manager was configured with when created
+func (r *rookTestInfraManager) GetRookPlatform() enums.RookPlatformType {
+	return r.platformType
+}
+
+//Wrapper method for executing docker commands
+//if method fails it will panic
+func (r *rookTestInfraManager) executeDockerCommand(containerId string, subCommand enums.DockerSubCommand , args ...string) (stdOut, stdErr string) {
+	var cmdPrefix string = ""
+ 	var cmdArgs []string
+
+	if subCommand != enums.Empty {
+		cmdPrefix = subCommand.String()
+	}
+
+	if strings.EqualFold(containerId, "") {
+		containerId = r.dockerContext.Get_ContainerId()
+	}
+
+	if subCommand == enums.Copy || subCommand == enums.Ps {
+		cmdArgs = append([]string{cmdPrefix}, args...)
+	} else {
+		cmdArgs = append([]string{cmdPrefix, containerId}, args...)
+	}
+
+	fmt.Println("Executing the command 'docker " + strings.Join(cmdArgs, " ") + "'")
+
+	stdOut, stdErr, err := r.dockerContext.Get_DockerClient().ExecuteCmd(cmdArgs)
+
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Println("Command succeeded...")
+
+	return stdOut, stdErr
+}
+
+//Get the k8s cluster creation script based on k8s version desired
+func getDindScriptName (k8sVersion enums.K8sVersion) (dindScriptName string) {
+	switch k8sVersion {
+	case enums.V1dot5:
+		dindScriptName = rookDindK8sClusterScriptv1_5
+	case enums.V1dot6:
+		dindScriptName = rookDindK8sClusterScriptv1_6
+	default:
+		panic(errors.New("Unsupported Kubernetes version: " + k8sVersion.String()))
+	}
+
+	return dindScriptName
+}
+
+//This method stands up a k8s cluster and pre-configures it for running tests
+func (r *rookTestInfraManager) ValidateAndSetupTestPlatform() {
+	dindScriptName := getDindScriptName(r.k8sVersion)
+
+	//make the k8s creation script executable
+	cmdOut := utils.ExecuteCommand(objects.Command_Args{Command:"chmod", CmdArgs:[]string{"+x", curdir + "/" + scriptsPath + "/" + dindScriptName}})
+
+	if cmdOut.Err != nil || cmdOut.ExitCode != 0 {
+		panic(errors.New("Failed to chmod cluster creation script"))
+	}
+
+	//launch the k8s creation script
+	cmdOut = utils.ExecuteCommand(objects.Command_Args{Command:  curdir + "/" + scriptsPath + "/" + dindScriptName, CmdArgs:[]string{ "up"}})
+
+	if cmdOut.Err != nil || cmdOut.ExitCode != 0 {
+		panic(errors.New("Failed to  cluster creation execute script"))
+	}
+
+	//Untaint all nodes making them schedulable
+	k8sClient := transport.CreateNewk8sTransportClient()
+
+	_, _, err := k8sClient.ExecuteCmd([]string{"taint", "nodes", "--all", "dedicated-"})
+
+	if err != nil && !strings.EqualFold(err.Error(), k8sFalsePostiveSuccessErrorMsg) {
+		panic(err)
+	}
+
+	//Drain node 2
+	_, _, err = k8sClient.ExecuteCmd([]string{"drain", "kube-node-2", "--force", "--ignore-daemonsets"})
```

Comment:
instead of draining and deleting the unnecessary node2 (which actually still leaves the container running, which is confusing), is it possible to update the DinD script to not even start node2 in the first place?